### PR TITLE
jobs: try using HEAD in clone_refs to use default remote branch

### PR DESCRIPTION
**Pull request by spiffxp**
trying this out per: https://github.com/kubernetes/test-infra/issues/20667#issuecomment-794308744

### CODE REVIEW

File: config/jobs/kubernetes/wg-k8s-infra/trusted/wg-k8s-infra-trusted.yaml [hunk 1]
```diff
@@ -22,6 +22,29 @@ periodics:
...
+  spec:
+    serviceAccountName: gsuite-groups-manager
+    containers:
+    - name: groups
+      image: golang:1.13
+      command:
+      - bash
+      args:
+      - -c
+      - "cd groups && make run -- --confirm"
```
Comment (BenTheElder) [inline config/jobs/kubernetes/wg-k8s-infra/trusted/wg-k8s-infra-trusted.yaml:47]: one thing that isn't clear just looking at the PR is the other intent of the job vs trying HEAD / why we need a new job here.

this looks like it is a copy+paste of a "production" job just with HEAD?
are there no issues running both?

**Comment by BenTheElder** [comment]
https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-k8sio-groups-head/1372989680978497536

> Init container initupload not ready: (state: terminated, reason: "Error", message: "ds.json\",\"file\":\"prow/pod-utils/gcs/upload.go:116\",\"func\":\"k8s.io/test-infra/prow/pod-utils/gcs.upload.func1\",\"level\":\"info\",\"msg\":\"Finished upload\",\"severity\":\"info\",\"time\":\"2021-03-19T19:13:48Z\"}\n{\"component\":\"initupload\",\"dest\":\"logs/ci-k8sio-groups-head/1372989680978497536/clone-log.txt\",\"file\":\"prow/pod-utils/gcs/upload.go:116\",\"func\":\"k8s.io/test-infra/prow/pod-utils/gcs.upload.func1\",\"level\":\"info\",\"msg\":\"Finished upload\",\"severity\":\"info\",\"time\":\"2021-03-19T19:13:48Z\"}\n{\"component\":\"initupload\",\"dest\":\"logs/ci-k8sio-groups-head/1372989680978497536/build-log.txt\",\"file\":\"prow/pod-utils/gcs/upload.go:116\",\"func\":\"k8s.io/test-infra/prow/pod-utils/gcs.upload.func1\",\"level\":\"info\",\"msg\":\"Finished upload\",\"severity\":\"info\",\"time\":\"2021-03-19T19:13:48Z\"}\n{\"component\":\"initupload\",\"dest\":\"logs/ci-k8sio-groups-head/latest-build.txt\",\"file\":\"prow/pod-utils/gcs/upload.go:116\",\"func\":\"k8s.io/test-infra/prow/pod-utils/gcs.upload.func1\",\"level\":\"info\",\"msg\":\"Finished upload\",\"severity\":\"info\",\"time\":\"2021-03-19T19:13:48Z\"}\n{\"component\":\"initupload\",\"dest\":\"logs/ci-k8sio-groups-head/1372989680978497536/started.json\",\"file\":\"prow/pod-utils/gcs/upload.go:116\",\"func\":\"k8s.io/test-infra/prow/pod-utils/gcs.upload.func1\",\"level\":\"info\",\"msg\":\"Finished upload\",\"severity\":\"info\",\"time\":\"2021-03-19T19:13:48Z\"}\n{\"component\":\"initupload\",\"dest\":\"logs/ci-k8sio-groups-head/1372989680978497536/finished.json\",\"file\":\"prow/pod-utils/gcs/upload.go:116\",\"func\":\"k8s.io/test-infra/prow/pod-utils/gcs.upload.func1\",\"level\":\"info\",\"msg\":\"Finished upload\",\"severity\":\"info\",\"time\":\"2021-03-19T19:13:48Z\"}\n{\"component\":\"initupload\",\"file\":\"prow/gcsupload/run.go:63\",\"func\":\"k8s.io/test-infra/prow/gcsupload.Options.Run\",\"level\":\"info\",\"msg\":\"Finished upload to blob storage\",\"severity\":\"info\",\"time\":\"2021-03-19T19:13:48Z\"}\n{\"component\":\"initupload\",\"error\":\"cloning the appropriate refs failed\",\"file\":\"prow/cmd/initupload/main.go:45\",\"func\":\"main.main\",\"level\":\"fatal\",\"msg\":\"Failed to initialize job\",\"severity\":\"fatal\",\"time\":\"2021-03-19T19:13:48Z\"}\n") Init container place-entrypoint not ready: (state: waiting, reason: "PodInitializing", message: "")

**Comment by BenTheElder** [comment]
> HEAD is now at 94928697 Merge pull request #1809 from chewong/promote-agnhost-2-30
> $ git branch --force HEAD FETCH_HEAD
> fatal: 'HEAD' is not a valid branch name.

File: config/jobs/kubernetes/wg-k8s-infra/trusted/wg-k8s-infra-trusted.yaml (hunk 1 above)
Comment (spiffxp) [reply config/jobs/kubernetes/wg-k8s-infra/trusted/wg-k8s-infra-trusted.yaml:47]: there are not, and yes that was the intent, I wanted a repo that had its default branch renamed

**Pull request merged**